src/builtins.c

```diff
@@ -416,17 +419,29 @@ static uintptr_t NOINLINE jl_object_id__cold(jl_datatype_t *dt, jl_value_t *v) J
 
 JL_DLLEXPORT inline uintptr_t jl_object_id_(jl_value_t *tv, jl_value_t *v) JL_NOTSAFEPOINT
 {
+    uintptr_t *hash_value = (uintptr_t*)ptrhash_bp(&object_table, v);
+    if(*hash_value != (uintptr_t)HT_NOTFOUND) 
+        return *hash_value;
```

Comment:
It is very confusing to call a pointer a value. I suggest naming it `hash_pointer`, or `hashp` for short.
```suggestion
    uintptr_t *hashp = (uintptr_t*)ptrhash_bp(&object_table, v);
    uintptr_t hash = *hashp;
    if (hash != (uintptr_t)HT_NOTFOUND)
        return hash;
```